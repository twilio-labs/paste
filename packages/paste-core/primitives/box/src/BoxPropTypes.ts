import * as PropTypes from 'prop-types';
import {
  isPaddingTokenProp,
  isMarginTokenProp,
  isWidthTokenProp,
  isMinWidthTokenProp,
  isMaxWidthTokenProp,
  isHeightTokenProp,
  isMinHeightTokenProp,
  isMaxHeightTokenProp,
  isBackgroundColorTokenProp,
  isBorderRadiusTokenProp,
  isBorderColorTokenProp,
  isBorderWidthTokenProp,
  isBoxShadowTokenProp,
  ResponsiveProp,
} from '@twilio-paste/style-props';

export const BoxPropTypes = {
  as: PropTypes.string as any,
  // Generic
  cursor: ResponsiveProp(PropTypes.string),
  content: ResponsiveProp(PropTypes.string),
  appearance: ResponsiveProp(PropTypes.string),
  animation: ResponsiveProp(PropTypes.string),
  transform: ResponsiveProp(PropTypes.string),
  transformOrigin: ResponsiveProp(PropTypes.string),
  visibility: ResponsiveProp(PropTypes.string),
  whiteSpace: ResponsiveProp(PropTypes.string),
  userSelect: ResponsiveProp(PropTypes.string),
  pointerEvents: ResponsiveProp(PropTypes.string),
  boxSizing: ResponsiveProp(PropTypes.string),
  resize: ResponsiveProp(PropTypes.string),
  transition: ResponsiveProp(PropTypes.string),
  listStyleType: ResponsiveProp(PropTypes.string),
  listStylePosition: ResponsiveProp(PropTypes.string),
  listStyleImage: ResponsiveProp(PropTypes.string),
  objectFit: ResponsiveProp(PropTypes.string),
  objectPosition: ResponsiveProp(PropTypes.string),
  backgroundAttachment: ResponsiveProp(PropTypes.string),
  outline: ResponsiveProp(PropTypes.string),
  float: ResponsiveProp(PropTypes.string),
  willChange: ResponsiveProp(PropTypes.string),
  textDecoration: ResponsiveProp(PropTypes.string),
  // layout
  width: isWidthTokenProp,
  minWidth: isMinWidthTokenProp,
  maxWidth: isMaxWidthTokenProp,
  height: isHeightTokenProp,
  minHeight: isMinHeightTokenProp,
  maxHeight: isMaxHeightTokenProp,
  size: isWidthTokenProp,
  display: ResponsiveProp(PropTypes.string),
  verticalAlign: ResponsiveProp(PropTypes.string),
  overflow: ResponsiveProp(PropTypes.string),
  overflowX: ResponsiveProp(PropTypes.string),
  overflowY: ResponsiveProp(PropTypes.string),
  // position
  position: ResponsiveProp(PropTypes.string),
  top: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  right: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  bottom: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  left: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  zIndex: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  // space
  margin: isMarginTokenProp,
  marginTop: isMarginTokenProp,
  marginRight: isMarginTokenProp,
  marginBottom: isMarginTokenProp,
  marginLeft: isMarginTokenProp,
  padding: isPaddingTokenProp,
  paddingTop: isPaddingTokenProp,
  paddingRight: isPaddingTokenProp,
  paddingBottom: isPaddingTokenProp,
  paddingLeft: isPaddingTokenProp,
  // background
  background: ResponsiveProp(PropTypes.string),
  backgroundColor: isBackgroundColorTokenProp,
  backgroundImage: ResponsiveProp(PropTypes.string),
  backgroundSize: ResponsiveProp(PropTypes.string),
  backgroundPosition: ResponsiveProp(PropTypes.string),
  backgroundRepeat: ResponsiveProp(PropTypes.string),
  // border
  borderRadius: isBorderRadiusTokenProp,
  borderBottomLeftRadius: isBorderRadiusTokenProp,
  borderBottomRightRadius: isBorderRadiusTokenProp,
  borderTopLeftRadius: isBorderRadiusTokenProp,
  borderTopRightRadius: isBorderRadiusTokenProp,
  borderWidth: isBorderWidthTokenProp,
  borderBottomWidth: isBorderWidthTokenProp,
  borderLeftWidth: isBorderWidthTokenProp,
  borderRightWidth: isBorderWidthTokenProp,
  borderTopWidth: isBorderWidthTokenProp,
  borderColor: isBorderColorTokenProp,
  borderBottomColor: isBorderColorTokenProp,
  borderLeftColor: isBorderColorTokenProp,
  borderRightColor: isBorderColorTokenProp,
  borderTopColor: isBorderColorTokenProp,
  borderStyle: ResponsiveProp(PropTypes.string),
  borderTopStyle: ResponsiveProp(PropTypes.string),
  borderRightStyle: ResponsiveProp(PropTypes.string),
  borderBottomStyle: ResponsiveProp(PropTypes.string),
  borderLeftStyle: ResponsiveProp(PropTypes.string),
  // shadow
  boxShadow: isBoxShadowTokenProp,
  // flexbox
  alignItems: ResponsiveProp(PropTypes.string),
  alignContent: ResponsiveProp(PropTypes.string),
  justifyItems: ResponsiveProp(PropTypes.string),
  justifyContent: ResponsiveProp(PropTypes.string),
  flexWrap: ResponsiveProp(PropTypes.string),
  flexDirection: ResponsiveProp(PropTypes.string),
  flex: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  flexGrow: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  flexShrink: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  flexBasis: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  justifySelf: ResponsiveProp(PropTypes.string),
  alignSelf: ResponsiveProp(PropTypes.string),
  order: ResponsiveProp(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
};
