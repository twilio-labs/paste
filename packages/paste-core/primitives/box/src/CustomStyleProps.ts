export const customStyleProps = {
  color: {
    property: 'color',
    scale: 'textColors',
  },
  backgroundColor: {
    property: 'backgroundColor',
    scale: 'backgroundColors',
  },
  borderColor: {
    property: 'borderColor',
    scale: 'borderColors',
  },
  borderBottomColor: {
    property: 'borderBottomColor',
    scale: 'borderColors',
  },
  borderLeftColor: {
    property: 'borderLeftColor',
    scale: 'borderColors',
  },
  borderRightColor: {
    property: 'borderRightColor',
    scale: 'borderColors',
  },
  borderTopColor: {
    property: 'borderTopColor',
    scale: 'borderColors',
  },
  content: true,
  cursor: true,
  appearance: true,
  transition: true,
  transform: true,
  animation: true,
  transformOrigin: true,
  visibility: true,
  userSelect: true,
  pointerEvents: true,
  boxSizing: true,
  resize: true,
  listStyleType: true,
  listStylePosition: true,
  listStyleImage: true,
  objectFit: true,
  objectPosition: true,
  outline: true,
  float: true,
  willChange: true,
  clip: true,
  backgroundAttachment: true,
  textAlign: true,
  textTransform: true,
  textDecoration: true,
  textOverflow: true,
  whiteSpace: true,
  wordBreak: true,
  wordWrap: true,
  overflowWrap: true,
  opacity: true,
  borderCollapse: true,
  borderSpacing: true,
  tableLayout: true,
  fontVariantNumeric: true,
  columnGap: {
    property: 'columnGap',
    scale: 'space',
  },
  rowGap: {
    property: 'rowGap',
    scale: 'space',
  },
};
