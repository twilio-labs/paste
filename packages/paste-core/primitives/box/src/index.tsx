import styled from '@emotion/styled';
import {compose, layout, space, background, border, boxShadow, position, flexbox, system} from 'styled-system';
import css from '@styled-system/css';
import {
  LayoutProps,
  SpaceProps,
  BackgroundProps,
  BorderProps,
  ShadowProps,
  PositionProps,
  FlexboxProps,
  TypographyProps,
} from '@twilio-paste/style-props';
import {
  CursorProperty,
  AppearanceProperty,
  AnimationProperty,
  TransformProperty,
  TransformOriginProperty,
  VisibilityProperty,
  WhiteSpaceProperty,
  UserSelectProperty,
  PointerEventsProperty,
  BoxSizingProperty,
  ResizeProperty,
  TransitionProperty,
  ListStyleTypeProperty,
  ListStylePositionProperty,
  ListStyleImageProperty,
  ObjectFitProperty,
  ObjectPositionProperty,
  BackgroundAttachmentProperty,
  OutlineProperty,
  FloatProperty,
  WillChangeProperty,
} from 'csstype';
import {PseudoPropStyles} from './PseudoPropStyles';
import {BoxPropTypes} from './BoxPropTypes';

export interface BaseBoxProps
  extends React.HTMLAttributes<any>,
    LayoutProps,
    SpaceProps,
    BackgroundProps,
    BorderProps,
    ShadowProps,
    PositionProps,
    FlexboxProps {
  as?: keyof JSX.IntrinsicElements;
  content?: string;
  cursor?: CursorProperty;
  appearance?: AppearanceProperty;
  animation?: AnimationProperty;
  transform?: TransformProperty;
  transformOrigin?: TransformOriginProperty<string>;
  visibility?: VisibilityProperty;
  whiteSpace?: WhiteSpaceProperty;
  userSelect?: UserSelectProperty;
  pointerEvents?: PointerEventsProperty;
  boxSizing?: BoxSizingProperty;
  resize?: ResizeProperty;
  transition?: TransitionProperty;
  listStyleType?: ListStyleTypeProperty;
  listStylePosition?: ListStylePositionProperty;
  listStyleImage?: ListStyleImageProperty;
  objectFit?: ObjectFitProperty;
  objectPosition?: ObjectPositionProperty<string>;
  backgroundAttachment?: BackgroundAttachmentProperty;
  outline?: OutlineProperty<string>;
  float?: FloatProperty;
  willChange?: WillChangeProperty;
  textDecoration?: TypographyProps['textDecoration'];
  /** Typed as any because Box can literally be any HTML element */
  ref?: any | null;
}

interface PseudoStylesProps {
  _after?: BaseBoxProps;
  _before?: BaseBoxProps;
  _focus?: BaseBoxProps;
  _hover?: BaseBoxProps;
  _active?: BaseBoxProps;
  _pressed?: BaseBoxProps;
  _selected?: BaseBoxProps;
  _focusWithin?: BaseBoxProps;
  _invalid?: BaseBoxProps;
  _disabled?: BaseBoxProps;
  _grabbed?: BaseBoxProps;
  _expanded?: BaseBoxProps;
  _checked?: BaseBoxProps;
  _mixed?: BaseBoxProps;
  _odd?: BaseBoxProps;
  _even?: BaseBoxProps;
  _visited?: BaseBoxProps;
  _readOnly?: BaseBoxProps;
  _first?: BaseBoxProps;
  _last?: BaseBoxProps;
  _groupHover?: BaseBoxProps;
  _notFirst?: BaseBoxProps;
  _notLast?: BaseBoxProps;
  _placeholder?: BaseBoxProps;
}

export interface BoxProps extends BaseBoxProps, PseudoStylesProps {}

const extraConfig = system({
  backgroundColor: {
    property: 'backgroundColor',
    scale: 'backgroundColors',
  },
  borderColor: {
    property: 'borderColor',
    scale: 'borderColors',
  },
  animation: true,
  appearance: true,
  transform: true,
  transformOrigin: true,
  visibility: true,
  whiteSpace: true,
  userSelect: true,
  pointerEvents: true,
  boxSizing: true,
  cursor: true,
  resize: true,
  transition: true,
  listStyleType: true,
  listStylePosition: true,
  listStyleImage: true,
  objectFit: true,
  objectPosition: true,
  backgroundAttachment: {
    property: 'backgroundAttachment',
  },
  outline: true,
  float: true,
  willChange: true,
  textDecoration: true,
});

const getPseudoStyles = (props: BoxProps): {} => {
  const pseudoProps = Object.keys(props).filter(propName => propName.startsWith('_'));

  if (pseudoProps.length === 0) {
    return {};
  }

  const pseudoStyles = {};
  pseudoProps.forEach(pseudoProp => {
    if (PseudoPropStyles[pseudoProp] != null) {
      pseudoStyles[PseudoPropStyles[pseudoProp]] = props[pseudoProp];
    }
  });

  return css(pseudoStyles);
};

/* eslint-disable emotion/syntax-preference */
export const Box = styled.div(
  {
    boxSizing: 'border-box',
  },
  compose(
    space,
    layout,
    flexbox,
    background,
    border,
    boxShadow,
    position,
    extraConfig
  ),
  getPseudoStyles
) as React.FC<BoxProps>;
/* eslint-enable */

Box.displayName = 'Box';

if (process.env.NODE_ENV === 'development') {
  Box.propTypes = BoxPropTypes;
}

export * from './SafelySpreadProps';
