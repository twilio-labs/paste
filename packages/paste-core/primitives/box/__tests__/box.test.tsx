import * as React from 'react';
import renderer from 'react-test-renderer';
import {Theme} from '@twilio-paste/theme';
import {Box} from '../src';

describe('Backgrounds', () => {
  it('should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box backgroundColor="colorBackgroundPrimary">background single</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box backgroundColor={['colorBackgroundPrimaryLight', 'colorBackgroundPrimary']}>background responsive</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Borders', () => {
  it('should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box
            borderStyle="solid"
            borderColor="colorBorderPrimaryDark"
            borderWidth="borderWidth10"
            borderRadius="borderRadius20"
          >
            border single
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box
            borderStyle={['dashed', 'dotted', 'solid']}
            borderColor={['colorBorderPrimaryDark', 'colorBorderPrimaryLight']}
            borderWidth={['borderWidth10', 'borderWidth20']}
            borderRadius={['borderRadius0', 'borderRadius10']}
          >
            border responsive
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Sizes', () => {
  it('should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box width="size10" minWidth="size0" maxWidth="size20" height="size10" minHeight="size0" maxHeight="size20">
            size single
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box
            width={['size10', 'size20']}
            minWidth={['size0', 'size10']}
            maxWidth={['size20', 'size30']}
            height={['size10', 'size20']}
            minHeight={['size0', 'size10']}
            maxHeight={['size20', 'size30']}
          >
            size responsive
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Spaces', () => {
  it('(A) it should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box margin="space20">space single 1</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('(A) it should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box margin={['space20', 'space30']}>space responsive 1</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('(B) it should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box marginTop="space20" marginRight="space20" marginBottom="space30" marginLeft="space30">
            space single 2
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('(B)it should render responsive values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box
            marginTop={['space20', 'space30']}
            marginRight={['space20', 'space30']}
            marginBottom={['space30', 'space40']}
            marginLeft={['space30', 'space40']}
          >
            space responsive 2
          </Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Shadows', () => {
  it('should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box boxShadow="shadowCard">shadow single</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box boxShadow={['shadowCard', 'shadowFocus']}>shadow responsive</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ZIndex', () => {
  it('should render single values', (): void => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box zIndex="zIndex10">z-index single</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render responsive values', () => {
    const tree = renderer
      .create(
        <Theme.Provider theme="console">
          <Box zIndex={['zIndex10', 'zIndex20']}>z-index responsive</Box>
        </Theme.Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('Pseudo-class props', () => {
    it('should generate pseudo-class CSS', (): void => {
      const tree = renderer
        .create(
          <Theme.Provider theme="console">
            <Box
              _hover={{padding: 'space10'}}
              _active={{padding: 'space10'}}
              _focus={{padding: 'space10'}}
              _visited={{padding: 'space10'}}
              _even={{padding: 'space10'}}
              _odd={{padding: 'space10'}}
              _disabled={{padding: 'space10'}}
              _checked={{padding: 'space10'}}
              _mixed={{padding: 'space10'}}
              _selected={{padding: 'space10'}}
              _invalid={{padding: 'space10'}}
              _pressed={{padding: 'space10'}}
              _readOnly={{padding: 'space10'}}
              _first={{padding: 'space10'}}
              _last={{padding: 'space10'}}
              _expanded={{padding: 'space10'}}
              _grabbed={{padding: 'space10'}}
              _notFirst={{padding: 'space10'}}
              _notLast={{padding: 'space10'}}
              _groupHover={{padding: 'space10'}}
              _before={{padding: 'space10'}}
              _after={{padding: 'space10'}}
              _focusWithin={{padding: 'space10'}}
              _placeholder={{padding: 'space10'}}
            >
              PseudoBox
            </Box>
          </Theme.Provider>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
