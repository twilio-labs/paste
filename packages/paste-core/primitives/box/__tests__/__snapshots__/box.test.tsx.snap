// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backgrounds it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #b3d3e9;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    background-color: #027ac5;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    background responsive
  </div>
</div>
`;

exports[`Backgrounds it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  background-color: #027ac5;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    background single
  </div>
</div>
`;

exports[`Borders it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  border-style: dashed;
  border-color: #003e82;
  border-width: 1px;
  border-radius: 0;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    border-style: dotted;
    border-color: #b3d3e9;
    border-width: 2px;
    border-radius: 3px;
  }
}

@media screen and (min-width:64rem) {
  .emotion-0 {
    border-style: solid;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    border responsive
  </div>
</div>
`;

exports[`Borders it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  border-style: solid;
  border-color: #003e82;
  border-width: 1px;
  border-radius: 4px;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    border single
  </div>
</div>
`;

exports[`Shadows it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  box-shadow: 0 2px 4px 0 rgba(40,42,43,0.3);
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    box-shadow: 0 0 0 4px rgba(2,122,197,0.8);
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    shadow responsive
  </div>
</div>
`;

exports[`Shadows it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  box-shadow: 0 2px 4px 0 rgba(40,42,43,0.3);
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    shadow single
  </div>
</div>
`;

exports[`Sizes it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  width: 5.5rem;
  min-width: 0;
  max-width: 12rem;
  height: 5.5rem;
  min-height: 0;
  max-height: 12rem;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    width: 12rem;
    min-width: 5.5rem;
    max-width: 18.5rem;
    height: 12rem;
    min-height: 5.5rem;
    max-height: 18.5rem;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
    height={
      Array [
        "size10",
        "size20",
      ]
    }
    width={
      Array [
        "size10",
        "size20",
      ]
    }
  >
    size responsive
  </div>
</div>
`;

exports[`Sizes it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  width: 5.5rem;
  min-width: 0;
  max-width: 12rem;
  height: 5.5rem;
  min-height: 0;
  max-height: 12rem;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
    height="size10"
    width="size10"
  >
    size single
  </div>
</div>
`;

exports[`Spaces (A) it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  margin: 0.25rem;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    margin: 0.5rem;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    space responsive 1
  </div>
</div>
`;

exports[`Spaces (A) it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  margin: 0.25rem;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    space single 1
  </div>
</div>
`;

exports[`Spaces (B) it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  margin-top: 0.25rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;
  margin-left: 0.5rem;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    space single 2
  </div>
</div>
`;

exports[`Spaces (B)it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  margin-top: 0.25rem;
  margin-right: 0.25rem;
  margin-bottom: 0.5rem;
  margin-left: 0.5rem;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    margin-top: 0.5rem;
    margin-right: 0.5rem;
    margin-bottom: 0.75rem;
    margin-left: 0.75rem;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    space responsive 2
  </div>
</div>
`;

exports[`ZIndex Pseudo-class props it should generate pseudo-class CSS 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
}

.emotion-0:hover {
  padding: 0.125rem;
}

.emotion-0:active,
.emotion-0[data-active=true] {
  padding: 0.125rem;
}

.emotion-0:focus {
  padding: 0.125rem;
}

.emotion-0:visited {
  padding: 0.125rem;
}

.emotion-0:nth-of-type(even) {
  padding: 0.125rem;
}

.emotion-0:nth-of-type(odd) {
  padding: 0.125rem;
}

.emotion-0:disabled,
.emotion-0:disabled:focus,
.emotion-0:disabled:hover,
.emotion-0[aria-disabled=true],
.emotion-0[aria-disabled=true]:focus,
.emotion-0[aria-disabled=true]:hover {
  padding: 0.125rem;
}

.emotion-0[aria-checked=true] {
  padding: 0.125rem;
}

.emotion-0[aria-checked=mixed] {
  padding: 0.125rem;
}

.emotion-0[aria-selected=true] {
  padding: 0.125rem;
}

.emotion-0[aria-invalid=true] {
  padding: 0.125rem;
}

.emotion-0[aria-pressed=true] {
  padding: 0.125rem;
}

.emotion-0[aria-readonly=true],
.emotion-0[readonly] {
  padding: 0.125rem;
}

.emotion-0:first-of-type {
  padding: 0.125rem;
}

.emotion-0:last-of-type {
  padding: 0.125rem;
}

.emotion-0[aria-expanded=true] {
  padding: 0.125rem;
}

.emotion-0[aria-grabbed=true] {
  padding: 0.125rem;
}

.emotion-0:not(:first-of-type) {
  padding: 0.125rem;
}

.emotion-0:not(:last-of-type) {
  padding: 0.125rem;
}

[role=group]:hover .emotion-0 {
  padding: 0.125rem;
}

.emotion-0:before {
  padding: 0.125rem;
}

.emotion-0:after {
  padding: 0.125rem;
}

.emotion-0:focus-within {
  padding: 0.125rem;
}

.emotion-0::-webkit-input-placeholder {
  padding: 0.125rem;
}

.emotion-0::-moz-placeholder {
  padding: 0.125rem;
}

.emotion-0:-ms-input-placeholder {
  padding: 0.125rem;
}

.emotion-0::placeholder {
  padding: 0.125rem;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    PseudoBox
  </div>
</div>
`;

exports[`ZIndex it should render responsive values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  z-index: 10;
}

@media screen and (min-width:25rem) {
  .emotion-0 {
    z-index: 20;
  }
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    z-index responsive
  </div>
</div>
`;

exports[`ZIndex it should render single values 1`] = `
.emotion-2 {
  font-family: 'Whitney SSm A','Whitney SSm B','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.875rem;
  line-height: 1.5rem;
  color: #282a2b;
  font-weight: 400;
}

.emotion-2 *,
.emotion-2 *::after,
.emotion-2 *::before {
  box-sizing: border-box;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  z-index: 10;
}

<div
  className="emotion-2"
>
  <div
    className="emotion-0 emotion-1"
  >
    z-index single
  </div>
</div>
`;
